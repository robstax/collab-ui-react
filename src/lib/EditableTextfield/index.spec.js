import React from 'react';
import { shallow, mount } from 'enzyme';
import { EditableTextfield } from '@collab-ui/react';

describe('tests for <EditableTextfield />', () => {
  it('should render an Editable Textfield', () => {
    const wrapper = shallow(<EditableTextfield />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render one Editable Textfield', () => {
    const container = shallow(<EditableTextfield />);

    expect(container.find('.cui-editable-textfield__button').length).toEqual(1);
    expect(container.find('Input').length).toEqual(0);
  });

  it('should click on Editable Textfield', () => {
    const container = shallow(<EditableTextfield inputText='Hello World' />);
    container.find('.cui-editable-textfield__button').simulate('click');

    expect(container.find('.cui-editable-textfield__button').length).toEqual(0);
    expect(container.find('Input').length).toEqual(1);
  });

  it('should pass buttonClassName to button', () => {
    const container = shallow(<EditableTextfield buttonClassName='buttonClass' className='inputClass' />);

    expect(container.find('.inputClass').exists()).toEqual(false);
    expect(container.find('.buttonClass').exists()).toEqual(true);
  });

  it('should pass className to Input', () => {
    const container = shallow(<EditableTextfield buttonClassName='buttonClass' className='inputClass' />);
    container.find('.cui-editable-textfield__button').simulate('click');

    expect(container.find('.buttonClass').exists()).toEqual(false);
    expect(container.find('.inputClass').exists()).toEqual(true);
  });

  it('should pass props to Input', () => {
    const container = shallow(<EditableTextfield title='testProp' />);
    container.find('.cui-editable-textfield__button').simulate('click');

    expect(container.find('Input').props().title).toEqual('testProp');
  });

  it('should pass buttonProps to button', () => {
    const container = shallow(<EditableTextfield buttonProps={{ title: 'testProp' }} />);

    expect(container.find('.cui-editable-textfield__button').props().title).toEqual('testProp');
  });

  it('should pass alignment prop to container', () => {
    const container = shallow(<EditableTextfield alignment='center' />);

    expect(container.find('.cui-editable-textfield--center').exists()).toEqual(true);
  });

  it('should click on Editable Textfield, edit and save', () => {
    const container = mount(<EditableTextfield inputText={'Hello World'}/>);
    container.find('.cui-editable-textfield__button').simulate('click');

    container.find('.cui-input').simulate('change', { target: { value: "Hello World Tom" } });
    container.find('.cui-input').simulate('keydown', { keyCode: 13, key: 'Enter' });

    expect(container.find('.cui-editable-textfield__button').text()).toEqual('Hello World Tom');
    expect(container.find('.cui-editable-textfield__button').length).toEqual(1);
    expect(container.find('Input').length).toEqual(0);
  });

  it('should click on Editable Textfield, edit and exit(escape)', () => {
    const container = mount(<EditableTextfield inputText={'Hello World'}/>);
    container.find('.cui-editable-textfield__button').simulate('click');

    container.find('.cui-input').simulate('change', { target: { value: "Hello World Tom" } });
    container.find('.cui-input').simulate('keydown', { keyCode: 27, key: 'Escape' });

    expect(container.find('.cui-editable-textfield__button').text()).toEqual('Hello World');
    expect(container.find('.cui-editable-textfield__button').length).toEqual(1);
    expect(container.find('Input').length).toEqual(0);
  });

  it('should click on Editable Textfield, edit and blur', () => {
    const container = mount(<EditableTextfield inputText={'Hello World'}/>);
    container.find('.cui-editable-textfield__button').simulate('click');

    container.find('.cui-input').simulate('change', { target: { value: "Hello World Tom" } });
    container.find('.cui-input').simulate('blur');

    expect(container.find('.cui-editable-textfield__button').text()).toEqual('Hello World Tom');
    expect(container.find('.cui-editable-textfield__button').length).toEqual(1);
    expect(container.find('Input').length).toEqual(0);
  });

  it('should handle empty value', () => {
    const container = mount(<EditableTextfield inputText={'Hello World'}/>);
    container.find('.cui-editable-textfield__button').simulate('click');

    container.find('.cui-input').simulate('change', { target: { value: "" } });
    container.find('.cui-input').simulate('blur');

    expect(container.find('.cui-editable-textfield__button').text()).toEqual('');
    expect(container.find('.cui-editable-textfield__button').length).toEqual(1);
    expect(container.find('Input').length).toEqual(0);
  });

});
