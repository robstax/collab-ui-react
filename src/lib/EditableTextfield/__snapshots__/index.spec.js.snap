// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <EditableTextfield /> should render an Editable Textfield 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditableTextfield
    alignment="left"
    buttonClassName=""
    buttonProps={null}
    className=""
    disabled={false}
    handleDoneEditing={null}
    inputText=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="cui-editable-textfield__button"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          \u00A0
        </div>,
      ],
      "className": "cui-editable-textfield cui-editable-textfield--left",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "\u00A0",
          "className": "cui-editable-textfield__button",
          "onClick": [Function],
          "onKeyPress": [Function],
          "role": "button",
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": "\u00A0",
        "type": "div",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="cui-editable-textfield__button"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            \u00A0
          </div>,
        ],
        "className": "cui-editable-textfield cui-editable-textfield--left",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "\u00A0",
            "className": "cui-editable-textfield__button",
            "onClick": [Function],
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": "\u00A0",
          "type": "div",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
